src/main/org/codehaus/groovy/reflection/CachedField.java

```diff
@@ -51,6 +52,12 @@ public int getModifiers() {
      */
     public Object getProperty(final Object object) {
         try {
+            AccessPermissionChecker.checkAccessPermission(field);
+        }
+        catch (AccessControlException ex) {
+            throw new IllegalArgumentException("Illegal access to field" + " " + field.getName());
```

Comment:
To use GroovyRuntimeException we need to patch also MetaClassImpl.getProperty . It was not possible with byte buddy, but as a groovy patch it is possible. I shall do it in a separate commit.